src/Main.c

```diff
@@ -131,61 +117,50 @@ int comparePerformance(){
+void outputPerformance(char name[], int* array_of_inputs[], int sizes[], int length) {
+  // TODO: make useable unit for ticks
+  clock_t total_t, total_t_ascending, total_t_descending;
   printf("----------%s------------\n", name);
   printf("elements\tduration:\trandom\tascending\tdescending\n" );
   for (int i = 0; i<length; i++){
-  start_t = clock();
-  sort_function(array_of_inputs[i],sizes[i]);
-  end_t = clock();
-  total_t = (double)(end_t - start_t);
+    // sorted random
+    total_t = measureSort(name, array_of_inputs[i], sizes[i]);
 
-  // time if already ascending sorted array
-  start_t = clock();
-  sort_function(array_of_inputs[i],sizes[i]);
-  end_t = clock();
-  total_t_ascending = (double)(end_t - start_t);
+    // already sorted ascending
+    total_t_ascending = measureSort(name, array_of_inputs[i], sizes[i]);
 
-  bubbleSortDescending(array_of_inputs[i], sizes[i]);
+    // make sort descending
+    bubbleSortDescending(array_of_inputs[i], sizes[i]);
 
-   // time if already descending sorted array
-  start_t = clock();
-  sort_function(array_of_inputs[i],sizes[i]);
-  end_t = clock();
-  total_t_descending = (double)(end_t - start_t);
+    total_t_descending = measureSort(name, array_of_inputs[i], sizes[i]);
 
-  printf("%d\t\t\t\t%lu\t%lu\t\t%lu\n", sizes[i], total_t, total_t_ascending, total_t_descending);
+    printf("%d\t\t\t\t%lu\t%lu\t\t%lu\n", sizes[i], total_t, total_t_ascending, total_t_descending);
   }
 }
 
-// for sort function that take array, start index and end index as parameter
-void outputPerformance2(char name[], int* array_of_inputs[], int sizes[], int length, int (*sort_function)(int[], int, int)){
-  clock_t start_t, end_t, total_t, start_t_ascending, end_t_ascending, total_t_ascending, start_t_descending, end_t_descending, total_t_descending;
-  printf("----------%s------------\n", name);
-  printf("elements\tduration:\trandom\tascending\tdescending\n" );
-  for (int i = 0; i<length; i++){
+clock_t measureSort(char name[], int array_of_inputs[], int length) {
+  clock_t start_t, end_t, total_t;
   start_t = clock();
-  sort_function(array_of_inputs[i],0,sizes[i]-1);
+  executeSort(name, array_of_inputs, length);
   end_t = clock();
   total_t = (double)(end_t - start_t);
+  return total_t;
+}
 
-  // time if already ascending sorted array
-  start_t_ascending = clock();
-  sort_function(array_of_inputs[i],0,sizes[i]-1);
-  end_t_ascending = clock();
-  total_t_ascending = (double)(end_t_ascending - start_t_ascending);
-
-  bubbleSortDescending(array_of_inputs[i], sizes[i]);
-
-   // time if already descending sorted array
-  start_t_descending = clock();
-  sort_function(array_of_inputs[i],0,sizes[i]-1);
-  end_t_descending = clock();
-  total_t_descending = (double)(end_t_descending - start_t_descending);
-
-  printf("%d\t\t\t\t%lu\t%lu\t\t%lu\n", sizes[i], total_t, total_t_ascending, total_t_descending);
-  }
+int executeSort(char name[], int arr[], int length) {
+  if (strcmp(name,"bub")) {
```

Comment:
if i'm not getting this wrong: bub is the name of the char array containing the string "Bubblesort"
so should it not be (strcmp(name,"Bubblesort") == 0) 
OR without the "" : (strcmp(name, bub) == 0) (if this is possible, because it means comparing the string to itself)
strcmp takes two char pointers as arguments and returns 0 if the strings match, >0 if the first non-matching character in str1 is greater (in ASCII) than that of str2, <0 if the first non-matching character in str1 is greater (in ASCII) than that of str2.
https://www.programiz.com/c-programming/library-function/string.h/strcmp

Reply:
yess totally right - thanks!
and comparison with the variable is much nicer, too
(though I thought that the if statement would evaluate the function return value and if 0 execute, but whatever)